azure-kusto-data/azure/kusto/data/_models.py

```diff
@@ -48,31 +50,29 @@ def __init__(self, columns, row):
 
-            if column_type in ["datetime", "timespan", "decimal"]:
-                if value is None:
-                    typed_value = None
-                    if HAS_PANDAS:
-                        self._hidden_values.append(None)
-                else:
-                    # If you are here to read this, you probably hit some datetime/timedelta inconsistencies.
-                    # Azure-Data-Explorer(Kusto) supports 7 decimal digits, while the corresponding python types supports only 6.
-                    # One example why one might want this precision, is when working with pandas.
-                    # In that case, use azure.kusto.data.helpers.dataframe_from_result_table which takes into account the original value.
-                    typed_value = KustoResultRow.convertion_funcs[column_type](value)
-
-                    # this is a special case where plain python will lose precision, so we keep the precise value hidden
-                    # when transforming to pandas, we can use the hidden value to convert to precise pandas/numpy types
-                    if HAS_PANDAS:
-                        if column_type == "datetime":
-                            self._hidden_values.append(to_pandas_datetime(value))
-                        if column_type == "timespan":
-                            self._hidden_values.append(to_pandas_timedelta(value, typed_value))
-                        if column_type == "decimal":
-                            self._hidden_values.append(value)
+            if value is None:
+                typed_value = None
+                if HAS_PANDAS:
+                    self._hidden_values.append(None)
             else:
-                typed_value = value
+                # If you are here to read this, you probably hit some datetime/timedelta inconsistencies.
+                # Azure-Data-Explorer(Kusto) supports 7 decimal digits, while the corresponding python types supports only 6.
+                # One example why one might want this precision, is when working with pandas.
+                # In that case, use azure.kusto.data.helpers.dataframe_from_result_table which takes into account the original value.
+                typed_value = (
+                    KustoResultRow.convertion_funcs[column_type](value)
+                    if column_type in KustoResultRow.convertion_funcs
+                    else value
+                )
+
+                # This is a special case where plain python will lose precision, so we keep the precise value hidden.
+                # When transforming to pandas, we can use the hidden value to convert to precise pandas/numpy types
                 if HAS_PANDAS:
-                    self._hidden_values.append(value)
+                    self._hidden_values.append(
+                        KustoResultRow.pandas_funcs[column_type](value, typed_value)
+                        if column_type in KustoResultRow.pandas_funcs
+                        else value
```

Comment:
shouldn't this `else` be `typed_value`?

Reply:
[Here](https://github.com/Azure/azure-kusto-python/blob/e626d40cdb079d717d05133b9bde301bb071e73b/azure-kusto-data/azure/kusto/data/_models.py#L71) and [here](https://github.com/Azure/azure-kusto-python/blob/6bf07c3c82c00ccb6ec2bf52711e0723dfbf71de/azure-kusto-data/azure/kusto/data/_models.py#L73) value is appended (which is a string representing decimal). Should it append typed_value?